de.tudarmstadt.ukp.wikipedia.api/src/it/java/de/tudarmstadt/ukp/wikipedia/api/PerformanceIT.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Technische Universität Darmstadt under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The Technische Universität Darmstadt 
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.wikipedia.api;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+
+import de.tudarmstadt.ukp.wikipedia.api.exception.WikiApiException;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.fail;
+
+public class PerformanceIT implements WikiConstants {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static Wikipedia wiki;
+    private static int retrievedNumberOfPages;
+
+    // The system under test
+    private static PerformanceTest pt;
+
+    @BeforeClass
+    public static void setupWikipedia() {
+        Properties configuration = loadConfiguration();
+        retrievedNumberOfPages = Integer.parseInt(configuration.getProperty("performance.pages.retrieved"));
+        DatabaseConfiguration dbConfig = obtainITDBConfiguration(configuration);
+        try {
+            wiki = new Wikipedia(dbConfig);
+            int maxiCycles = Integer.parseInt(configuration.getProperty("performance.cycles.maxi"));
+            int pageCycles = Integer.parseInt(configuration.getProperty("performance.cycles.page"));
+            pt = new PerformanceTest(wiki, maxiCycles, pageCycles);
+        } catch (Exception e) {
+            fail("Wikipedia could not be initialized: "+e.getLocalizedMessage());
```

Comment:
Just throw the exception. This "eats" the stack trace.

Reply:
will fix